engine/database/transaction.go

```diff
@@ -1,8 +1,27 @@
 package database
 
 // Transaction represents a transaction in the database.
-// Fixme: This is not integrated with out db engine yet.
 type Transaction struct {
 	ID       string `gorm:"type:char(64);primaryKey"`
 	TxStatus TxStatus
+
+	Outputs []*Output `gorm:"foreignKey:TxID"`
+	Data    []*Data   `gorm:"foreignKey:TxID"`
+}
+
+// TableName implements gorm.Tabler to override automatic table naming.
+// NOTE: This is because we have already a legacy table named "transactions".
+// TODO: Remove this when we have migrated all data.
+func (t *Transaction) TableName() string {
+	return "new_transactions"
+}
+
+// AddOutputs adds outputs to the transaction.
+func (t *Transaction) AddOutputs(outputs ...*Output) {
+	t.Outputs = append(t.Outputs, outputs...)
```

Comment:
:thinking_face: would it really work if we're adding inputs and outputs as outputs?
I mean:
- we have tx1 with output_1 which has primary key (tx1,0)
- we have tx2 with input from output_1, and with output on index 0, 
How it will be stored and connected?